### PR TITLE
Added support for oauth2 authentication type.

### PR DESCRIPTION
**Pull request by taylorsteffanj**

### CODE REVIEW

File: ibmpairs/paw.py [hunk 1]
```diff
@@ -498,15 +513,20 @@ def __init__(
         else:
             self.publish2GUI    = False
         # set GUI password (default to PAIRS API password)
-        self.guiPassword        = guiPassword if guiPassword is not None else self.auth[1]
+        if type(auth) is authentication.OAuth2:
```
Comment (elljoh) [inline ibmpairs/paw.py]: `auth` may be `None` here when `self.auth` is set on Line 493

**Comment by cmalbrec** [comment]
We will resolve https://travis-ci.com/github/IBM/ibmpairs/jobs/482758997 by sunset of support for Python 2 which reached its end-of-life more than a year ago (https://www.python.org/doc/sunset-python-2/).

File: ibmpairs/paw.py (hunk 1 above)
Comment (cmalbrec) [reply ibmpairs/paw.py]: we fix by `auth` to `self.auth` in line 516

File: tests/test_paw.py [hunk 2]
```diff
@@ -123,24 +127,33 @@
...
-            PAIRS_SERVER, PAIRS_BASE_URI, PAIRS_USER, PAIRS_PASSWORD_FILE_NAME,
-        )
-    )
+if AUTH_TYPE.lower() in ['api-key', 'apikey', 'api key']:
+    OAUTH = authentication.OAuth2(host         = PAIRS_SERVER,
+                                  username     = PAIRS_USER,
+                                  api_key_file = PAIRS_PASSWORD_FILE_NAME
+                                 )
+    PAIRS_JWT_TOKEN = OAUTH.jwt_token
+    PAIRS_CREDENTIALS       = (PAIRS_USER, PAIRS_PASSWORD,)
```
Comment (cmalbrec) [inline tests/test_paw.py:136]: it might be useful to add some logging to track testing

**Review by cmalbrec — APPROVED** [review]
modification to existing code `paw.py` and its tests `test_paw.py` reviewed for backward compatibility

**Pull request merged**